gapic/templates/tests/unit/gapic/%name_%version/%sub/test_macros.j2

```diff
@@ -887,6 +887,50 @@ def test_{{ method_name }}_rest(request_type):
     {% if not field.oneof or field.proto3_optional %}
     {# ignore oneof fields that might conflict with sample_request #}
     request_init["{{ field.name }}"] = {{ field.merged_mock_value(method.http_options[0].sample_request(method).get(field.name)) }}
+    # The version of a generated dependency at test runtime may differ from the version used during generation
+    # Delete any fields which are not present in the current runtime dependency
+    # See https://github.com/googleapis/gapic-generator-python/issues/1748
+    if hasattr({{ method.input.ident }}.meta.fields["{{ field.name }}"].message, "DESCRIPTOR"):
+        subfields_to_delete = []
+
+        # Get all subfields for the message
+        nested_fields = [
+            (field.name, subfield.name)
+            for field in {{ method.input.ident }}.meta.fields["{{ field.name }}"].message.DESCRIPTOR.fields
+            if field.message_type
+            for subfield in field.message_type.fields
+        ]
+
+        # For each item in the sample request, create a list of sub fields which are not present at runtime
+        for field, value in request_init["{{ field.name }}"].items():
+            result = None
+            is_repeated = False
+            # For repeated fields
+            if isinstance(value, list) and len(value):
+                is_repeated = True
+                result = value[0]
+            # For fields where the type is another message
+            if isinstance(value, dict):
+                result = value
+
+            if result:
+                for subfield in result.keys():
+                    if (field, subfield) not in nested_fields:
+                        subfields_to_delete.append(
+                            {"field": field, "subfield": subfield, "is_repeated": is_repeated}
+                        )
+
+        # Remove fields from the sample request which are not present in the runtime version of the dependency
+        for subfield_to_delete in subfields_to_delete:
+            if subfield_to_delete.get("subfield"):
+                if subfield_to_delete.get("is_repeated"):
+                    del request_init["{{ field.name }}"][subfield_to_delete.get("field")][0][
```

Comment:
This is only deleting `subfield_to_delete` in the first item of the `repeated`, right? What about subsequent items? We may have a test that populates more than one item in the `repeated` field

Reply:
Can you share an example of what this would look like in a proto? We're already have a fragment test with a repeated field. It's the one in `tests/fragments/google/api/service.proto`
```
  // A list of all proto message types included in this API service.
  // Types referenced directly or indirectly by the `apis` are automatically
  // included.  Messages which are not referenced but shall be included, such as
  // types used by the `google.protobuf.Any` type, should be listed here by
  // name by the configuration author. Example:
  //
  //     types:
  //     - name: google.protobuf.Int32
  repeated google.protobuf.Type types = 4;

  // A list of all enum types included in this API service.  Enums referenced
  // directly or indirectly by the `apis` are automatically included.  Enums
  // which are not referenced but shall be included should be listed here by
  // name by the configuration author. Example:
  //
  //     enums:
  //     - name: google.someapi.v1.SomeEnum
  repeated google.protobuf.Enum enums = 5;
```

Reply:
Let's say `books` is a repeated field. If the test populates `books[0]` and `books[1]`, and we encounter the problematic field in `books[0]`, then don't we have to delete that field from `books[0]` and `books[1]` both, i.e. from all elements of `book`?